src/ua.c

```diff
@@ -932,6 +935,192 @@ int ua_uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 }
 
 
+static bool uri_only_user(const struct uri *uri)
+{
+	bool ret;
+	struct sa sa;
+
+	/* Note:
+	 * If only user is given then uri_decode sets uri->host instead of
+	 * uri->user. We don't know if this is a bug. But if somebody changes
+	 * this behavior then the following line has to be adapted.          */
+	ret = pl_isset(&uri->host) && !pl_isset(&uri->user);
+
+	/* exclude IP addresses */
+	ret = ret && 0 != sa_set(&sa, &uri->host, 0);
+
+	return ret;
+}
+
+
+static bool uri_user_and_host(const struct uri *uri)
+{
+	bool ret;
+
+	ret = pl_isset(&uri->host) && pl_isset(&uri->user);
+
+	return ret;
+}
+
+
+static bool uri_host_local(const struct uri *uri)
+{
+	const struct sa *sal;
+	struct sa sap;
+	int err;
+
+	if (!uri)
+		return false;
+
+	if (!pl_strcmp(&uri->host, "localhost"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "127.0.0.1"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "::1"))
+		return true;
+
+	sal = net_laddr_af(baresip_network(), AF_INET);
+
+	err = sa_set(&sap, &uri->host, 0);
+	if (err)
+		return false;
+
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+
+#ifdef HAVE_INET6
+	sal = net_laddr_af(baresip_network(), AF_INET6);
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+#endif
+
+	return false;
+}
+
+
+static bool acc_inactive(const struct account *acc)
```

Comment:
consider inversing the negated logic here: acc_is_active()

I find it much easier to read code that has correct logic (i.e. not negated).